### PR TITLE
Fix: Nullable reward epoch received in + cleanup

### PR DESCRIPTION
**Pull request by rhyslbw**
:pushpin: **fix!: Reward.receivedIn nullability**
Rewards are known before being "paid out", so not all rewards
have been received. This change fixes the model to ensure the
upcoming rewards have a nullable relationship to the epoch
received in.

:pushpin: **refactor!: remove remnants of previously deprecated config**
Two class constructor functions have now been updated to reflect
the deprecation of using the last configured major Cardano version,
which were missed in the previous breaking change. There are no
changes to the service config, so it will only impact library users.

### CODE REVIEW

**Review by mkazlauskas — APPROVED** [review]

File: packages/api-cardano-db-hasura/schema.graphql [hunk 1]
```diff
@@ -611,7 +611,7 @@ type Reward {
   address: StakeAddress!
   amount: String!
   earnedIn: Epoch!
-  receivedIn: Epoch!
+  receivedIn: Epoch
```
Comment (iadmytro) [inline packages/api-cardano-db-hasura/schema.graphql:614]: Could you add test for the case when receivedIn is null/undefined?

**Review by iadmytro — CHANGES_REQUESTED** [review]

File: packages/api-cardano-db-hasura/schema.graphql (hunk 1 above)
Comment (rhyslbw) [reply packages/api-cardano-db-hasura/schema.graphql:614]: Not at the moment, unfortunately, as we only have API tests against _testnet_ where we cannot control this state. It' 's a transient state, so apps just need to guard against it being undefined, which is all this change does.

**Review by iadmytro — APPROVED** [review]
